src/AppSystem/Launcher.vala

```diff
@@ -328,4 +353,24 @@ public class Dock.Launcher : BaseItem {
             multiple_windows_open = app.windows.length > 1;
         }
     }
+
+    private void on_enter (Gtk.EventControllerMotion controller, double event_x, double event_y) {
+        if (!timeout_is_running) {
+            Timeout.add_once (POPOVER_NAME_TIMEOUT, show_popover_name);
```

Comment:
There is an issue with Timeout.add_once, it doen't ref the object, so if the object gets destroyed while the timeout is running, it will crash. I think you can replace `timeout_is_running` with timeout id and check whether it's 0 to decide if timeout's running. And then you can do `Source.remove (timeout_id)` in the destructor to prevent the crash.

Reply:
Okay, I think I changed it to what you suggested. There is another instance of `Timeout.add_once` already present in the file that is not using this technique. Is this other instance not affected by this issue or should this be adjusted in the future as well?

Reply:
Oops, yes, that definitely got overlooked.